server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/jsieve/delivery/SieveExecutor.java

```diff
@@ -114,7 +119,14 @@ private SieveFactory createFactory(Log log) throws MessagingException {
     public boolean execute(MailAddress recipient, Mail mail) throws MessagingException {
         Preconditions.checkNotNull(recipient, "Recipient for mail to be spooled cannot be null.");
         Preconditions.checkNotNull(mail.getMessage(), "Mail message to be spooled cannot be null.");
-
+        Optional<Attribute> oSieveNotification = mail.getAttribute(SIEVE_NOTIFICATION);
+        if (oSieveNotification.isPresent()) {
+            AttributeValue<Boolean> attrValue = (AttributeValue<Boolean>) oSieveNotification.get().getValue();
+            if (attrValue.value()) {
+                throw new MessagingException("Do not process Sieve error notification emails");
```

Comment:
I would rather `return false;`

After all, the sieve notifications are valid messages we should handle.